controllers/designer/kaoto_controller_action_rbac.go

```diff
@@ -2,17 +2,133 @@ package designer
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/kaotoIO/kaoto-operator/pkg/apply"
+
+	"github.com/kaotoIO/kaoto-operator/pkg/client"
+
+	"github.com/kaotoIO/kaoto-operator/apis/designer/v1alpha1"
+
+	"github.com/go-logr/logr"
+	"github.com/kaotoIO/kaoto-operator/pkg/pointer"
 	"github.com/pkg/errors"
-	"go.uber.org/multierr"
 	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	ctrlcl "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	rbacv1ac "k8s.io/client-go/applyconfigurations/rbac/v1"
+
+	"go.uber.org/multierr"
 	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func NewRBACAction() Action {
+	return &rbacAction{
+		l: ctrl.Log.WithName("action-rbac"),
+	}
+}
+
 type rbacAction struct {
+	l logr.Logger
+}
+
+func (a *rbacAction) Cleanup(ctx context.Context, rr *ReconciliationRequest) error {
+
+	//
+	// Some resources cannot be garbage collected and automatically removed using ownership as i.e. they
+	// are cluster scoped such as a ClusterRoleBinding hence must be deleted
+	//
+
+	err := rr.Client.RbacV1().ClusterRoleBindings().Delete(ctx, rr.Kaoto.Namespace+"-"+rr.Kaoto.Name, metav1.DeleteOptions{
+		PropagationPolicy: pointer.Any(metav1.DeletePropagationForeground),
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failure removing ClusterRoleBindings %s-%s", rr.Kaoto.Namespace, rr.Kaoto.Name)
+	}
+
+	return nil
+}
+
+func (a *rbacAction) Configure(_ context.Context, c *client.Client, b *builder.Builder) (*builder.Builder, error) {
+
+	b = b.Owns(&corev1.ServiceAccount{}, builder.WithPredicates(
+		predicate.Or(
+			predicate.ResourceVersionChangedPredicate{},
+		)))
+
+	b = b.Watches(
```

Comment:
Is this watcher needed for deleting the CRB or does it serve a different purpose?

Reply:
The need of an explicit watcher versus `Builder.Own` is because the `ClusterRoleBinding` is a non namespaced resource, hence watching by ownership is not supported. 

Watching the  `ClusterRoleBinding` is mainly to detect external changes to the resource and revert them.